src/error.rs

```diff
@@ -122,12 +130,20 @@ impl From<UserError> for Error {
 
 impl fmt::Display for Error {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        use self::Kind::*;
-
         match self.kind {
-            Proto(ref reason) => write!(fmt, "protocol error: {}", reason),
-            User(ref e) => write!(fmt, "user error: {}", e),
-            Io(ref e) => fmt::Display::fmt(e, fmt),
+            Kind::Reset(_, reason, initiator) => {
+                write!(fmt, "stream reset by {}: {}", initiator, reason)
+            }
+            Kind::GoAway(ref debug_data, reason, initiator) => {
+                write!(fmt, "go away from {}: {}", initiator, reason)?;
+                if !debug_data.is_empty() {
+                    write!(fmt, " ({:?})", debug_data)?;
```

Comment:
Hm. Alternate output might be a good use here... Do you know what other HTTP/2 libraries do with the debug-data? Like netty, or grpc, or other...?

Reply:
[Netty](https://github.com/netty/netty/blob/201f6b6bb2dcdff07df6e30552b96fedbcf21759/codec-http2/src/main/java/io/netty/handler/codec/http2/StreamBufferingEncoder.java#L84-L112) doesn't format the related exception at all.

[grpc-java](https://github.com/grpc/grpc-java/blob/0838b736748b398292a696d2444f9aa0f6778081/netty/src/main/java/io/grpc/netty/NettyClientHandler.java#L282) seems to log debug data directly though.

IMO we should just always include it in Display otherwise most people won't benefit from it.

Reply:
@seanmonstar Let's merge it? 🥺